packages/typespec-autorest/src/openapi.ts

```diff
@@ -163,6 +163,8 @@ const defaultOptions = {
 
 export async function $onEmit(context: EmitContext<AutorestEmitterOptions>) {
   const resolvedOptions = { ...defaultOptions, ...context.options };
+  // set api-version to all to prevent tcgc versioning projection
+  (context.options as any)["api-version"] = "all";
```

Comment:
I do feel like this is quite hacky, if another emitter wanted that they'd have to do the same thing. Could we maybe instead have some options in `createSdkContext` to use or not to use the projection

Reply:
yeah, my only concern with that approach is now there are two places and ways to input this information. Would you be ok for now if we just use this code, open an issue to keep track of this and put it on the backlog, and if another library runs into this issue, we can brainstorm more?

Reply:
well I don't see it as 2 place to input the info. One is what the user ask for the other is what does your emitter support. With autorest it handles the versioning itself so that param is pointless. 

I think it would be better to have 
```ts
createSDKContext(context, {
   versionStrategy: "latest-or-api-version" | "ignore",
});
```

Reply:
if you think this is really not good then sure but doing what we have now is bypassing the type system so we could easily break it

Reply:
ok that seems fair, let me update the code here. Thanks!